### PR TITLE
issue382 MyWaveWAV OpenDAP mapper

### PR DESCRIPTION
**Pull request by korvinos**
A mapper for MyWaveWAV 4km wave model archive

### CODE REVIEW

File: nansat/mappers/mapper_opendap_mywave.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            raise WrongMapperError
+
+        self.create_vrt(filename, gdal_dataset, gdal_metadata, timestamp, ds, bands, cachedir)
+        mm = pti.get_gcmd_instrument('Computer')
+        ee = pti.get_gcmd_platform('MODELS')
+        self.dataset.SetMetadataItem('instrument', json.dumps(mm))
+        self.dataset.SetMetadataItem('platform', json.dumps(ee))
+        self.dataset.SetMetadataItem('Data Center', 'NO/MET')
+        self.dataset.SetMetadataItem('Entry Title', str(ds.getncattr('title')))
+
```
Comment (mortenwh) [inline nansat/mappers/mapper_opendap_mywave.py]: Some missing metadata should be added:

* `time_coverage_*`
* `ISO_topic_category`
* `set_metadata('gcmd_location', json.dumps(pti.get_gcmd_location('sea surface')))`

File: nansat/mappers/mapper_opendap_mywave.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            str, YYYY-mm-ddThh:MMZ
+
+        Examples
+        --------
+            >>> Mapper.get_date('/path/to/MyWave_wam4_WAVE_20171029T18Z.nc')
+            '2017-10-29T18:00Z'
+        """
+        _, filename = os.path.split(filename)
+        t = datetime.strptime(filename.split('_')[-1], '%Y%m%dT%HZ.nc')
+        return datetime.strftime(t, '%Y-%m-%dT%H:%MZ')
```
Comment (mortenwh) [inline nansat/mappers/mapper_opendap_mywave.py]: there is another way to do this, e.g., as in `envisat.py`:

```
from dateutil.parser import parse
...
self.dataset.SetMetadataItem('time_coverage_start', parse(gdalMetadata["SPH_FIRST_LINE_TIME"]).isoformat())
```

File: nansat/tests/mappers/test_mapper_opendap_mywave.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import unittest
+from nansat.mappers.mapper_opendap_mywave import Mapper
+from netCDF4 import Dataset
+
+
+class MyWaveOpenDAPTests(unittest.TestCase):
+
+    def setUp(self):
+        self.src = 'http://thredds.met.no/thredds/dodsC/fou-hi/mywavewam4archive' \
```
Comment (mortenwh) [inline nansat/tests/mappers/test_mapper_opendap_mywave.py]: This test package should not use real data but rather mocks. The goal is to add mapper testing to the continuous integration, and then accessing real data is too time consuming. Could you rather rewrite the tests using mocking, or move it to the integration tests (which are meant to be run locally)?

File: nansat/mappers/mapper_opendap_mywave.py (hunk 1 above)
Comment (korvinos) [reply nansat/mappers/mapper_opendap_mywave.py]: `time_coverage_*` metadata are added in opendap.py file during an execution of the `create_vrt` method

File: nansat/mappers/mapper_opendap_mywave.py (hunk 2 above)
Comment (korvinos) [reply nansat/mappers/mapper_opendap_mywave.py]: I believe that the method used in my code is much simpler and explicitly shows what happens

File: nansat/tests/mappers/test_mapper_opendap_mywave.py (hunk 3 above)
Comment (korvinos) [reply nansat/tests/mappers/test_mapper_opendap_mywave.py]: I test a parsing of the src (as a str) and do not exploit (I/O) the data from the link

File: nansat/mappers/mapper_opendap_mywave.py (hunk 2 above)
Comment (mortenwh) [reply nansat/mappers/mapper_opendap_mywave.py]: Maybe, but it is also repetition... Can you at least specify in the docstring and above line 69 that it is formatted in isoformat?

**Comment by korvinos** [comment]
I am not sure why the coverage increased by 1.2% after previous commits either. So not sure how to explain this.

**Pull request merged**